lib/net.js

```diff
@@ -319,14 +319,25 @@ Socket.prototype._onTimeout = function() {
 
 
 Socket.prototype.setNoDelay = function(enable) {
+  if (!this._handle) {
+    this.once('connect',
```

Comment:
Do EventEmitters guarantee the order of the calls?

```javascript
socket.setNoDelay(true);
socket.setNoDelay(false);
socket.setNoDelay(true);

socket.connect(..);
```

I don't see any guarantees in the docs.